kernel/work.c

```diff
@@ -597,6 +597,17 @@ static void work_queue_main(void *workq_ptr, void *p2, void *p3)
 			work = CONTAINER_OF(node, struct k_work, node);
 			flag_set(&work->flags, K_WORK_RUNNING_BIT);
 			flag_clear(&work->flags, K_WORK_QUEUED_BIT);
+			/* parasoft suppress item MISRAC2012-DIR_4_1-f "False-positive
+			 * In line 597, the work is figured out by CONTAINER_OF,
+			 * as a container of type struct k_work that contains the node.
+			 * The only way for it to be NULL is if node would be a member
+			 * of struct k_work object that has been placed at address NULL,
+			 * which should never happen, even line 634 ensures that.
```

Comment:
Using line numbers isn't either a great idea, comment will soon get out of context when file is updated. A good contextual information should be enough.